interop/tapir/src/test/scala/caliban/interop/tapir/TapirAdapterSpec.scala

```diff
@@ -159,45 +161,111 @@ object TapirAdapterSpec {
+                _           <- inputQueue.offer(
+                                 GraphQLWSInput(
+                                   Ops.Subscribe,
+                                   Some("id"),
+                                   Some(ObjectValue(Map("query" -> StringValue("subscription { characterDeleted }"))))
+                                 )
                                )
-                             )
-              sendDelete   = send(
-                               run((GraphQLRequest(Some("""mutation{ deleteCharacter(name: "Amos Burton") }""")), null))
-                             ).delay(3 seconds)
-              stop         = inputQueue.offer(GraphQLWSInput("stop", Some("id"), None))
-              messages    <- outputStream
-                               .tap(out => ZIO.when(out.`type` == "connection_ack")(sendDelete))
-                               .tap(out => ZIO.when(out.`type` == "data")(stop))
-                               .take(3)
-                               .runCollect
-                               .timeoutFail(new Throwable("timeout ws"))(30.seconds)
-                               .provideSomeLayer[SttpClient](Clock.live)
-            } yield messages
-
-          io.map { messages =>
-            assertTrue(messages.head.`type` == "connection_ack") &&
-            assertTrue(messages(1).payload.get.toString == """{"data":{"characterDeleted":"Amos Burton"}}""") &&
-            assertTrue(messages(2).`type` == "complete")
+                sendDelete   = send(
+                                 run((GraphQLRequest(Some("""mutation{ deleteCharacter(name: "Amos Burton") }""")), null))
+                               ).delay(3 seconds)
+                stop         = inputQueue.offer(GraphQLWSInput(Ops.Complete, Some("id"), None))
+                messages    <- outputStream
+                                 .tap(out =>
+                                   ZIO.whenCase(out) {
+                                     case Right(out) if out.`type` == Ops.ConnectionAck => sendDelete
+                                     case Right(out) if out.`type` == Ops.Next          => stop
+                                   }
+                                 )
+                                 .take(3)
+                                 .runCollect
+                                 .timeoutFail(new Throwable("timeout ws"))(60.seconds)
+                                 .provideSomeLayer[SttpClient](Clock.live)
+              } yield messages
+
+            io.map { messages =>
+              assertTrue(messages.head.map(_.`type`) == Right(Ops.ConnectionAck)) &&
+              assertTrue(
+                messages(1).map(_.payload.get.toString) == Right("""{"data":{"characterDeleted":"Amos Burton"}}""")
+              ) &&
+              assertTrue(messages(2).map(_.`type`) == Right(Ops.Complete))
+            }
           }
-        }
+        ) @@ TestAspect.sequential
```

Comment:
Do we need it if we have it below?

Reply:
I don't think so, I was trying to debug the tests and probably added it, but the issue turned out to be that the `ActorSystem` wasn't being shutdown in between tests.